### PR TITLE
fix: running `near_workspaces::compile_project` concurrently in tests

### PR DESCRIPTION
**Pull request by dj8yfo**
image

this is suppossed to address this issue

https://github.com/near/near-workspaces-rs/blob/main/workspaces/src/cargo/mod.rs#L35-L41

### CODE REVIEW

File: cargo-near-build/src/near/build/mod.rs
Comment (dj8yfo) [inline cargo-near-build/src/near/build/mod.rs:165]: sometimes the final destination file is written with non-`wasm-opt`-ed content, 
which might be the reason of failures in concurrent context when many tests run
`near_workspaces::compile_project`

Comment (dj8yfo) [reply cargo-near-build/src/near/build/mod.rs:165]: after introducing and intermediate stage `[project_name]_optimized.wasm`,  all tests will 
likely read from the same file (from `near_workspaces::compile_project`), which is equal to `[project_name]_optimized.wasm` , 

and not interchangeably from `[project_name].wasm` or `[project_name]_optimized.wasm`

Comment (dj8yfo) [reply cargo-near-build/src/near/build/mod.rs:165]: was able to easily reproduce this on this commit https://github.com/dj8yfo/near-sdk-rs/commit/027ce75a67a023432901c5eb2d70aa68a276c423

```bash
    receipts: [
        ExecutionOutcome {
            transaction_hash: 61H3aajosM8kDV3jbpuzFaAnMrTCb4DL871gVbbocNag,
            block_hash: HVzmo5HTPZJySPhG5CXMPvde5kMTAGigWKgREbttsoBH,
            logs: [],
            receipt_ids: [
                Dm63UmQr9mqrF6mDiRUxHTaWo8ESKbPHr1p9taEzSzaf,
            ],
            gas_burnt: NearGas {
                inner: 888148937360,
            },
            tokens_burnt: NearToken {
                inner: 88814893736000000000,
            },
            executor_id: AccountId(
                "dev-20241216183040-18506921751143",
            ),
            status: Failure(ActionError(ActionError { index: Some(0), kind: FunctionCallError(CompilationError(PrepareError(Deserialization))) })),
        },
    ],
    status: Failure(ActionError(ActionError { index: Some(0), kind: FunctionCallError(CompilationError(PrepareError(Deserialization))) })),
}
thread 'test_storage_deposit_refunds_excessive_deposit' panicked at tests/workspaces.rs:86:5:
assertion failed: res.is_success()


failures:
    simulate_transfer_call_when_called_contract_not_registered_with_ft
    simulate_transfer_call_with_promise_and_refund
    test_close_account_empty_balance
    test_storage_deposit_refunds_excessive_deposit

test result: FAILED. 9 passed; 4 failed; 0 ignored; 0 measured; 0 filtered out; finished in 34.43s
```

Comment (dj8yfo) [reply cargo-near-build/src/near/build/mod.rs:165]: adding https://github.com/near/near-sdk-rs/pull/1277/commits/ffab607525f02999ea82ed542ea051f2281d0dbd on top of previous change resolved flaky failures

Comment (dj8yfo) [reply cargo-near-build/src/near/build/mod.rs:165]: corresponding run in near-sdk ci - https://github.com/near/near-sdk-rs/actions/runs/12359514121

**Comment by dj8yfo** [comment]
@race-of-sloths include

**Comment by race-of-sloths** [comment]
@dj8yfo Thank you for your contribution! Your pull request is now a part of the [Race of Sloths](https://race-of-sloths.com)!
Weekly streak is on the road, smart strategy! Secure your streak with another PR!

[<picture>
    <source media="(prefers-color-scheme: dark)" srcset="https://race-of-sloths.com/images/bot-banner-dark.svg">
    <source media="(prefers-color-scheme: light)" srcset="https://race-of-sloths.com/images/bot-banner-light.svg">
    Shows inviting banner with latest news.
</picture>](https://race-of-sloths.com/latest-news)

[<picture>
    <source media="(prefers-color-scheme: dark)" srcset="https://badge.race-of-sloths.com/dj8yfo?type=bot&pr=near/cargo-near/266&theme=dark">
    <source media="(prefers-color-scheme: light)" srcset="https://badge.race-of-sloths.com/dj8yfo?type=bot&pr=near/cargo-near/266&theme=light">
    Shows profile picture for the author of the PR
</picture>](https://race-of-sloths.com/profile/dj8yfo)


<details><summary>Current status: <i>waiting for scoring</i></summary>

We're waiting for maintainer to score this pull request with `@race-of-sloths score [0,1,2,3,5,8,13]` command. Alternatively, autoscoring [1,2] will be applied for this pull request

</details><details><summary>What is the Race of Sloths</summary>
<p>
Race of Sloths is a friendly competition where you can participate in challenges and compete with other open-source contributors within your normal workflow

For contributors:

- Tag @race-of-sloths inside your pull requests
- Wait for the maintainer to review and score your pull request
- Check out your position in the [Leaderboard](https://race-of-sloths.com/leaderboard)
- Keep weekly and monthly streaks to reach higher positions
- Boast your contributions with a dynamic picture of your [Profile](https://race-of-sloths.com/profile/dj8yfo)

For maintainers:
- Score pull requests that participate in the Race of Sloths and receive a reward
- Engage contributors with fair scoring and fast responses so they keep their streaks
- Promote the Race to the point where the Race starts promoting you
- Grow the community of your contributors

Feel free to check [our website](https://race-of-sloths.com) for additional details!
</p>
</details> 

<details>
<summary>Bot commands</summary>

- For contributors
  - **Include a PR:** `@race-of-sloths include` to enter the Race with your PR
- For maintainers:
  - **Invite contributor** `@race-of-sloths invite` to invite the contributor to participate in a race or include it, if it's already a runner.
  - **Assign points:** `@race-of-sloths score [1/2/3/5/8/13]` to award points based on your assessment.
  - **Reject this PR:** `@race-of-sloths exclude` to send this PR back to the drawing board.
  - **Exclude repo:** `@race-of-sloths pause` to stop bot activity in this repo until `@race-of-sloths unpause` command is called

</details>

File: cargo-near-build/src/fs.rs
Comment (PolyProgrammist) [inline cargo-near-build/src/fs.rs:30]: Is there is some info about if file is wasm-opted? If so, we can check  just that instead of reading the whole file

File: cargo-near-build/src/fs.rs
Comment (PolyProgrammist) [inline cargo-near-build/src/fs.rs:26]: Does the method work with directories as well? Maybe use `if !from.is_file() || !to.is_file() -> return`

**Review by PolyProgrammist — COMMENTED** [review]
Here - should be near organisation https://github.com/near/near-sdk-rs/commit/ffab607525f02999ea82ed542ea051f2281d0dbd

**Comment by PolyProgrammist** [comment]
@race-of-sloths score 8

**Review by PolyProgrammist — APPROVED** [review]

**Pull request merged**
